File: python/101/tip_calc.py
subtotal = float(raw_input("Hi there! How much is your bill? "))
tipType = raw_input("How was your service? (good/fair/bad) ")
tip = 0.0
total = 0.0
if tipType == "good":
    tip = subtotal * 0.2
    total = subtotal * 1.2
elif tipType == "fair":
    tip = subtotal * 0.15
    total = subtotal * 1.15
else:
    tip = subtotal * 0.1
    total = subtotal * 1.10

print "Tip amount: $%.2f" % tip
print "Total amount: $%.2f" % total

File: javascript/101/tip_calc.js
function tipAmount(amount,service) {
  if (service === "good") {
    var tip = amount * 0.2;
  } else if (service === "fair") {
    var tip = amount * 0.15;
  } else {
    var tip = amount * 0.1;
  }
  return tip;
}

function totalAmount(amount,service) {
  if (service === "good") {
    amount += amount * 0.2;
  } else if (service === "fair") {
    amount += amount * 0.15;
  } else {
    amount += amount * 0.1;
  }
  return amount;
}

function splitTotal(amount,service,num) {
  total = totalAmount(amount,service);
  return total/num;
}

console.log(splitTotal(100,'good',5));
